### PR TITLE
Injimob 1204 share qr code

### PR DESCRIPTION
**Pull request by balachandarg-tw**
## Description

> Describe the Changes in the PR
- Updating the base64 string state variable from the getRef property of QRcode which gives base64 string.
- Using react-native-share , Passing the sharing options(text and image) to the library , so that it will be shared to other supported apps installed in device.

##[ Issue ticket number and link](https://mosip.atlassian.net/jira/software/c/projects/INJIMOB/boards/91?selectedIssue=INJIMOB-1204)

## Screenshots

> Add the Screenshots of the feature.

### CODE REVIEW

File: components/QrCodeOverlay.tsx [hunk 1]
```diff
@@ -103,6 +133,22 @@ export const QrCodeOverlay: React.FC<QrCodeOverlayProps> = props => {
                 backgroundColor={Theme.Colors.QRCodeBackgroundColor}
                 ecl={DEFAULT_ECL}
                 onError={onQRError}
+                getRef={data => (qrRef.current = data)}
+              />
+              <Button
+                testID="share"
+                margin="30 0 0 0"
```
Comment (sree96) [inline components/QrCodeOverlay.tsx:140]: Move this to themes

File: locales/ara.json [hunk 2]
```diff
@@ -82,7 +82,8 @@
         "title": "فشل حفظ بطاقة",
         "message": "حدث خطأ أثناء حفظ البطاقة في المتجر."
       }
-    }
+    },
+    "share_qr_code": "مشاركة رمز الاستجابة السريعة"
```
Comment (sree96) [inline locales/ara.json:86]: We use camelcase convention

File: locales/en.json [hunk 3]
```diff
@@ -82,7 +82,8 @@
         "title": "Failed to save the Card",
         "message": "Something went wrong while saving Card to the store."
       }
-    }
+    },
+    "share_qr_code": "Share QR Code"
```
Comment (sree96) [inline locales/en.json:86]: We use camelcase convention

File: locales/fil.json [hunk 4]
```diff
@@ -82,7 +82,8 @@
         "title": "Nabigong i-save ang Card",
         "message": "Nagkaproblema habang sine-save ang Card sa tindahan."
       }
-    }
+    },
+    "share_qr_code": "Ibahagi ang QR Code"
```
Comment (sree96) [inline locales/fil.json:86]: We use camelcase convention

File: locales/kan.json [hunk 5]
```diff
@@ -82,7 +82,8 @@
         "title": "ಕಾರ್ಡ್ ಉಳಿಸಲು ವಿಫಲವಾಗಿದೆ",
         "message": "ಸ್ಟೋರ್‌ನಲ್ಲಿ ಕಾರ್ಡ್ ಉಳಿಸುವಾಗ ಏನೋ ತಪ್ಪಾಗಿದೆ."
       }
-    }
+    },
+    "share_qr_code": "QR ಕೋಡ್ ಹಂಚಿಕೊಳ್ಳಿ"
```
Comment (sree96) [inline locales/kan.json:86]: We use camelcase convention

File: locales/tam.json [hunk 6]
```diff
@@ -82,7 +82,8 @@
         "title": "அட்டைஐ சேமிப்பதில் தோல்வி",
         "message": "கடையில் அட்டைஐ சேமிக்கும் போது ஏதோ தவறு ஏற்பட்டது."
       }
-    }
+    },
+    "share_qr_code": "QR குறியீட்டைப் பகிரவும்"
```
Comment (sree96) [inline locales/tam.json:86]: We use camelcase convention

File: shared/sharing/sharing-image-utils.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import RNShare, {ShareOptions} from 'react-native-share';
+
+export async function shareImageToAllSupportedApps(
+  sharingOptions: ShareOptions,
+): Promise<Boolean> {
+  try {
+    const shareStatusResult = await RNShare.open(sharingOptions);
+    console.log('Image shared suuccessfully::');
```
Comment (sree96) [inline shared/sharing/sharing-image-utils.ts:8]: Please remove the logs

**Review by sree96 — CHANGES_REQUESTED** [review]
Please address the comments

File: components/QrCodeOverlay.tsx (hunk 1 above)
Comment (balachandarg-tw) [reply components/QrCodeOverlay.tsx]: Done

File: locales/ara.json (hunk 2 above)
Comment (balachandarg-tw) [reply locales/ara.json]: Done

File: locales/fil.json (hunk 4 above)
Comment (balachandarg-tw) [reply locales/fil.json]: Done

File: locales/hin.json [hunk 8]
```diff
@@ -83,7 +83,8 @@
         "message1": "कार्ड को स्टोर में सेव करते समय कोई गड़बड़ी हुई.",
         "message": "स्टोर में कार्ड सहेजते समय कोई गड़बड़ी हुई।"
       }
-    }
+    },
+    "share_qr_code": "QR कोड साझा करें"
```
[the remark this answers is not in the document]
Comment (balachandarg-tw) [reply locales/hin.json]: Done

File: locales/kan.json (hunk 5 above)
Comment (balachandarg-tw) [reply locales/kan.json]: Done

File: locales/tam.json (hunk 6 above)
Comment (balachandarg-tw) [reply locales/tam.json]: Done

File: shared/sharing/sharing-image-utils.ts (hunk 7 above)
Comment (balachandarg-tw) [reply shared/sharing/sharing-image-utils.ts]: Removed

**Review by sree96 — APPROVED** [review]

File: components/QrCodeOverlay.tsx [hunk 9]
```diff
@@ -35,6 +37,34 @@ export const QrCodeOverlay: React.FC<QrCodeOverlayProps> = props => {
...
+  function updateBase64() {
+    qrRef.current.toDataURL(dataURL => {
+      setBase64String(`data:image/png;base64,${dataURL}`);
+    });
+    shareQRCode();
+  }
+
+  useEffect(() => {
+    shareQRCode();
+  }, [base64String]);
```
Comment (swatigoel) [inline components/QrCodeOverlay.tsx:53]: @balachandarg-tw @KiruthikaJeyashankar why we need shareQRCode in useEffect. Share QRCode is manual step by end user.

File: shared/sharing/sharing-image-utils.ts [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import RNShare, {ShareOptions} from 'react-native-share';
```
Comment (swatigoel) [inline shared/sharing/sharing-image-utils.ts:1]: Why this filename is in kebab case?

**Pull request merged**